tdms/src/iterator.cpp

```diff
@@ -246,33 +246,22 @@ using namespace std;
                            will be sampled, 1 means Ex, 2 Ey etc.
 */
 void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]){
-  
-  /*Local variables*/
-  #ifdef FDFLAG
-  bool useCD = true;  // Use central difference derivatives
-  #else
-  bool useCD = false; // Use pseudo-spectral derivatives
-  #endif
 
   auto params = SimulationParameters();
 
+  auto E_s = ElectricSplitField();
+  auto H_s = MagneticSplitField();
+  auto J_s = CurrentDensitySplitField();
+
   auto E = ElectricField();
   auto H = MagneticField();
+  auto E_copy = ElectricField();

```

Comment:
This seems odd out of context - leave a quick comment explaining why there's a copy?